tools/generator/dfg/sam/sam_device_tree.py

```diff
@@ -287,6 +294,20 @@ def driverOrder(e):
                     source.setAttributes("name", name, "value", source_id)
                 generators = driver.addChild("generators")
                 generators.setAttributes("value", str(p["gclk_data"]["generator_count"]))
+            # Add request data to DMA module
+            # Skip D09 devices, information is missing in raw data.
+            elif name in ("dmac", "xdmac", "pdc") and p["id"].series != "d09":
+                requests = driver.addChild("requests")
+                requests.addSortKey(lambda e: int(e["id"]))
```

Comment:
Oh wait, it's here… hmmm

Reply:
There is the same issue now for the DMA data:
```
+        <request device-variant="b" signal="ch0_tx" id="30" peripheral="dacc"/>
         <request device-series="e70|s70" device-pin="n" device-flash="21" device-variant="a" signal="tx" id="30" peripheral="dacc"/>
         <request device-series="e70|s70|v70" device-pin="j|n|q" device-flash="19|20" device-variant="a" signal="tx" id="30" peripheral="dacc"/>
         <request device-series="e70|s70|v71" device-pin="j|q" device-flash="20|21" device-variant="a" signal="tx" id="30" peripheral="dacc"/>
         <request device-series="v71" device-pin="n|q" device-flash="19" device-variant="a" signal="tx" id="30" peripheral="dacc"/>
-        <request device-variant="b" signal="ch0_tx" id="30" peripheral="dacc"/>
```

All these lines represent the entry `signal="ch0_tx" id="30" peripheral="dacc"`, but they differ in the filters added by the generator (`device-series`, `device-pin`, `device-flash`, `device-variant`). These would need to be included in the sort key somehow.

Reply:
The tree overlay algorithm is order deterministic, therefore there is no extra sort key necessary.
I refactored the requests a little (no need for a plural parent node) and added a more extensive sort key, seems to have fixed it. If you're happy please rebase the fixups.

Reply:
Thanks. I could have sworn I tested the same sort key change you added yesterday and it did not work for me. Must have messed something else up. I am happy it works now.